x-pack/plugins/observability_solution/observability_ai_assistant_app/public/functions/visualize_esql.test.tsx

```diff
@@ -142,4 +146,61 @@ describe('VisualizeESQL', () => {
       expect(screen.getByTestId('observabilityAiAssistantErrorsList')).toBeInTheDocument()
     );
   });
+
+  it('should not display the table on first render', async () => {
+    const lensService = {
+      ...lensPluginMock.createStartContract(),
+      stateHelperApi: jest.fn().mockResolvedValue({
+        formula: jest.fn(),
+        suggestions: jest.fn(),
+      }),
+    };
+    renderComponent({}, lensService);
+    // the button to render a table should be present
+    await waitFor(() =>
+      expect(
+        screen.getByTestId('observabilityAiAssistantLensESQLDisplayTableButton')
+      ).toBeInTheDocument()
+    );
```

Comment:
Another option: `screen.findByTestId`... has a built-in wait

Reply:
Cool y not https://github.com/elastic/kibana/pull/184106/commits/81354e2c7e482e9289b55e98d3558350c8209f26